public/app/features/live/dashboard/dashboardWatcher.ts

```diff
@@ -0,0 +1,166 @@
+      return;
+    }
+
+    // Check for changes
+    if (uid !== this.uid) {
+      this.leave();
+      this.channel = live.getChannel(LiveChannelScope.Grafana, 'dashboard', uid);
+      this.channel.getStream().subscribe(this.observer);
+      this.uid = uid;
+    }
+
+    console.log('Watch', uid);
+  }
+
+  leave() {
+    if (this.channel) {
+      this.channel.disconnect();
+    }
+    this.uid = undefined;
+  }
+
+  ignoreNextSave() {
+    this.ignoreSave = true;
+  }
+
+  observer = {
+    next: (event: LiveChannelEvent<DashboardEvent>) => {
+      // Send the editing state when connection starts
+      if (event.status && this.editing && event.status.state === LiveChannelConnectionState.Connected) {
+        this.sendEditingState();
+      }
+
+      if (event.message) {
+        if (event.message.sessionId === sessionId) {
+          return; // skip internal messages
+        }
+
+        const { action } = event.message;
+        switch (action) {
+          case DashboardEventAction.EditingStarted:
+          case DashboardEventAction.Saved: {
+            if (this.ignoreSave) {
+              this.ignoreSave = false;
+              return;
+            }
+
+            const dash = getDashboardSrv().getCurrent();
+            if (dash.uid !== event.message.uid) {
+              console.log('dashboard event for differnt dashboard?', event, dash);
+              return;
+            }
+
+            const changeTracker = getLegacyAngularInjector().get<any>('unsavedChangesSrv').tracker;
+            const showPopup = this.editing || changeTracker.hasChanges();
+
+            if (action === DashboardEventAction.Saved) {
+              if (showPopup) {
+                appEvents.emit(CoreEvents.showModalReact, {
+                  component: DashboardChangedModal,
+                  props: { event },
+                });
+              } else {
+                appEvents.emit(AppEvents.alertSuccess, ['Dashboard updated']);
+                this.reloadPage();
+              }
+            } else if (showPopup) {
+              if (action === DashboardEventAction.EditingStarted) {
+                appEvents.emit(AppEvents.alertWarning, [
+                  'Another session is editing this dashboard',
+                  event.message.message,
+                ]);
+              }
+            }
+            return;
+          }
+        }
+      }
+      console.log('DashboardEvent EVENT', event);
+    },
+  };
+
+  reloadPage() {
+    const $route = getLegacyAngularInjector().get<any>('$route');
+    if ($route) {
+      $route.reload();
+    } else {
+      location.reload();
+    }
+  }
+}
+
+export const dashboardWatcher = new DashboardWatcher();
+
+export function getDashboardChannelsFeature(): CoreGrafanaLiveFeature {
+  const dashboardConfig: LiveChannelConfig = {
+    path: '${uid}',
+    description: 'Dashboard change events',
+    variables: [{ value: 'uid', label: '${uid}', description: 'unique id for a dashboard' }],
```

Comment:
how are dashboardConfig.variables used?

Reply:
idea here is that this is mostly documentation for what variables exist and what they need to be.  Looking now, we should likely remove this until it has something concrete, but I want to be careful that the path based routing has *some* way to be more constrained